### PR TITLE
 Support save/load for lr_scheduler

### PR DESCRIPTION
**Pull request by wyg1997**
支持 WarmUpScheduler 保存自己以及内部 Scheduler 的状态（但不包括内部 Optimizer 的状态）

### CODE REVIEW

**Comment by wyg1997** [comment]
@L1aoXingyu 这里和昨天讨论的有一点不同，就是 WarmUpScheduler 不应该保存内部 Optimizer 的状态，因为他们不是一个类，save/load 之间不应该有交叉。而内部 Scheduler 是同一个基类对象，可以递归的一起 save/load。这里也是和 torch 的表现对齐。

File: python/oneflow/nn/optimizer/lr_scheduler.py [hunk 1]
```diff
@@ -119,3 +119,29 @@ def step(self):
             self._inner_lr_sch.step()
             # get right last_step from inner lr_scheduler
             self.last_step = self._inner_lr_sch.last_step
+
+    def state_dict(self):
+        """Returns the state of the scheduler as a :class:`dict`.
```
Comment (L1aoXingyu) [inline python/oneflow/nn/optimizer/lr_scheduler.py:124]: 是不是应该用 Return 而不是 Returns

File: python/oneflow/nn/optimizer/lr_scheduler.py [hunk 2]
```diff
@@ -119,3 +119,29 @@ def step(self):
...
+        """
+        state =  {
+            key: value for (key, value) in self.__dict__.items() if key != "_optimizer"
+        }
+        if self._inner_lr_sch is not None:
+            state["_inner_lr_sch"] = self._inner_lr_sch.state_dict()
+        return state
+
+    def load_state_dict(self, state_dict):
+        """Loads the schedulers state.
```
Comment (L1aoXingyu) [inline python/oneflow/nn/optimizer/lr_scheduler.py:134]: 同上

File: python/oneflow/nn/optimizer/lr_scheduler.py (hunk 1 above)
Comment (lixinqi) [reply python/oneflow/nn/optimizer/lr_scheduler.py:124]: 我记得是否要用第三人称单数这个问题再外网上也有人讨论。可以找找。

Comment (wyg1997) [reply python/oneflow/nn/optimizer/lr_scheduler.py:124]: https://www.python.org/dev/peps/pep-0257/

这里看道是没要求，但给的例子是用的动词原型，这里也用原型吧

**Review by L1aoXingyu — APPROVED** [review]

**Review by strint — APPROVED** [review]
